### PR TITLE
enable boot index for bootstrap VM

### PR DESCRIPTION
**Pull request by zshi-redhat**
It is expected by etcd and kubeapi in 4.4 deployment that baremetal(API) interface in bootstrap VM is the default interface (with interface order being 1). This commit sets boot index in bootstrap VM template based on OPENSHIFT_RHCOS_MAJOR_REL to ensure baremetal interface is the default one for 4.4 or later deployment.

### CODE REVIEW

File: terraform/cluster/bootstrap/templates/bootstrap_vm.tpl [hunk 1]
```diff
@@ -32,9 +34,32 @@
...
 <address type='pci' domain='0x0000' bus='0x00' slot='0x06' function='0x0'/>
 </disk>
+%{ if bootstrap_enable_boot_index == "true" ~}
+<interface type='bridge'>
+<mac address='52:54:00:82:68:3e'/>
+<source bridge='${baremetal_bridge}'/>
+<target dev='vnet0'/>
+<model type='rtl8139'/>
+<alias name='net0'/>
+<boot order='3'/>
```
Comment (abays) [inline terraform/cluster/bootstrap/templates/bootstrap_vm.tpl]: In your PR comment, it says:

```
It is expected by etcd and kubeapi in 4.4 deployment that baremetal(API) interface in bootstrap VM is the default interface (with interface order being 1).
```

It looks like here, though, the baremetal interface is being set to a higher index (3) than the provisioning interface index (2).  Should it be lower than the provisioning index to make sure it's the default?

Comment (zshi-redhat) [reply terraform/cluster/bootstrap/templates/bootstrap_vm.tpl]: There are two orders: one is boot order, configured via <boot order>, the other is interface order (when checking in OS with `ifconfig` or `ip link show`). provisioning interface needs to have lower boot order (hight boot priority) in order to provision the bootstrap VM from it.  baremetal interface needs to have lower interface order for etcd/kubeapi to get the right IP.

Comment (zshi-redhat) [reply terraform/cluster/bootstrap/templates/bootstrap_vm.tpl]: interface order is simply determinzed by which interface appear as the first in bootstrap VM xml.

Comment (abays) [reply terraform/cluster/bootstrap/templates/bootstrap_vm.tpl]: Ah, okay.  I think PCI address also matters as well, which I see you've done too (baremetal has a lower PCI address than provisioning when `bootstrap_enable_boot_index` is true).

Comment (zshi-redhat) [reply terraform/cluster/bootstrap/templates/bootstrap_vm.tpl]: right

**Pull request merged**